### ISSUE TITLE
Lua i/o model in 2.1.3

### ISSUE THREAD

**Issue by wolkstein** [opentx/opentx#2884]
hi,
simple question, is lua i/o model supported in 2.1.3? and if yes, how to use it in a way as opentx developers create the implementation.
i mean sd card structure(user space). maybe callback functions to noticed shutdown  and so on.
an example would be helpfully:).

thx
wolke

**Comment by bsongis** [comment]
It seems to me that I added it, yes. Did you try the standard i/o functions of the Lua manual?

**Comment by wolkstein** [comment]
not under 2.3.1. at least i test it under 2.0.17 without success. and than i use it as described in this reference. http://www.lua.org/pil/21.2.html

**Comment by projectkk2glider** [comment]
Original issue was #1471 and these are implemented IO functions:
```c
/*
** functions for 'io' library
*/
const luaL_Reg iolib[] = {
  {"close", io_close},
  // {"flush", io_flush},
  // {"input", io_input},
  // {"lines", io_lines},
  {"seek", io_seek},
  {"open", io_open},
  // {"output", io_output},
  // {"popen", io_popen},
  {"read", io_read},
  // {"tmpfile", io_tmpfile},
  // {"type", io_type},
  {"write", io_write},
  {NULL, NULL}
};
```

**Comment by projectkk2glider** [comment]
@wolkstein reopen this issue if it is still not working for you (after you study my comment above). You should try it first in the Companion simualtor and use the Debug window (F6) and add some `print()` statements to your Lua script.

**Comment by kilrah** [comment]
> not under 2.3.1. at least i test it under 2.0.17 without success.

It is only available in 2.1.

**Comment by wolkstein** [comment]
thx
